.github/workflows/self-hosted-e2e-tests.yml

```diff
@@ -0,0 +1,56 @@
+name: Self-hosted Sentry end to end tests
+on:
+  push:
+    branches:
+      - master
+      - releases/**
+  pull_request:
+
+# Cancel in progress workflows on pull_requests.
+# https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  self-hosted-end-to-end:
+    name: self-hosted tests
+    runs-on: ubuntu-20.04
+    # temporary, remove once we are confident the action is working
+    continue-on-error: true
+    timeout-minutes: 30
```

Comment:
should we set this lower? seems like 30 minutes is a bit long here, maybe 20?

Reply:
Unfortunately polling on the image can take a while, up to 10 minutes, and self-hosted CI itself takes about 10 minutes, so 20 minutes is cutting it too close. I wanted to err on the side of not causing timeout flakes. I imagine a few extra minutes if CI is stalled is not a huge problem.